server/model/index.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+// TODO: needs update when we integrate csv feature
+const dataReportSchema = schema.object({
+  saved_search_id: schema.string(),
+  time_range: schema.string(),
+  report_format: schema.oneOf([schema.literal('csv'), schema.literal('xlsx')]),
+});
+
+const visualReportSchema = schema.object({
+  url: schema.uri(),
+  window_width: schema.number({ defaultValue: 1200 }),
+  window_height: schema.number({ defaultValue: 800 }),
+  report_format: schema.oneOf([schema.literal('pdf'), schema.literal('png')]),
+});
+
+export const scheduleSchema = schema.object({
+  schedule_type: schema.oneOf([
+    schema.literal('Now'),
+    schema.literal('Future Date'),
```

Comment:
Sorry i missed these but "Date" in Future date and "Based" in Cron based are not capitalized either